src/Plugin/Field/FieldFormatter/LinkEmbedFormatter.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\url_embed\Plugin\Field\FieldFormatter\LinkEmbedFormatter.
+ */
+
+namespace Drupal\url_embed\Plugin\Field\FieldFormatter;
+
+use Drupal\Component\Utility\SafeMarkup;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\FormatterBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Url;
+use Drupal\link\LinkItemInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Embed\Embed;
+
+/**
+ * Plugin implementation of the 'URL Embed' formatter.
+ *
+ * @FieldFormatter(
+ *   id = "url_embed",
+ *   label = @Translation("Preview URL using Embed library"),
+ *   field_types = {
+ *     "link"
+ *   }
+ * )
+ */
+class LinkEmbedFormatter extends FormatterBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $plugin_id,
+      $plugin_definition,
+      $configuration['field_definition'],
+      $configuration['settings'],
+      $configuration['label'],
+      $configuration['view_mode'],
+      $configuration['third_party_settings']
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items) {
+    $elements = array();
+    foreach ($items as $delta => $item) {
+      $url = $item->getUrl() ?: Url::fromRoute('<none>');
+      $url_string = $url->toString();
+      $info = Embed::create($url_string);
+      $elements[$delta] = array(
+        '#markup' => SafeMarkup::set($info->code),
+      );
```

Comment:
No, because we want this to work with the <a href="https://www.drupal.org/project/fallback_formatter">Fallback Formatter module</a>, so it should be possible to use a mix of Link field type formatters. Embedding a URL is a much more discrete operation, so we should log when it fails.